packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -352,22 +388,47 @@ export class Canary extends cdk.Resource {
     };
   }
 
+  private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
+    if (!props.environmentVariables) {
+      return undefined;
+    }
+    return {
+      environmentVariables: props.environmentVariables,
+    };
+  }
+
   /**
-   * Returns a canary schedule object
-   */
+  * Returns a canary schedule object
+  */
   private createSchedule(props: CanaryProps): CfnCanary.ScheduleProperty {
     return {
       durationInSeconds: String(`${props.timeToLive?.toSeconds() ?? 0}`),
       expression: props.schedule?.expressionString ?? 'rate(5 minutes)',
     };
   }
 
-  private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+  private createVpcConfig(props: CanaryProps): CfnCanary.VPCConfigProperty | undefined {
```

Comment:
I think there's room to improve the customer experience here. From this [doc](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_VPC.html): 

```
If the VPC is not already enabled for internet access, you have two options:

    Enable it for internet access. For more information, see the following section [Giving internet access to your canary on a VPC](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_VPC.html#CloudWatch_Synthetics_VPC_Internet).

    If you want to keep your VPC private, you can configure the canary to send its data to CloudWatch and Amazon S3 through private VPC endpoints. If you have not already done so, you must create endpoints for these services on your VPC. For more information, see Using CloudWatch and CloudWatch Synthetics with interface VPC endpoints and [Amazon VPC Endpoints for Amazon S3](https://docs.aws.amazon.com/glue/latest/dg/vpc-endpoints-s3.html).
```

I wonder if we can check to make sure the subnet is public as a default, and provide an option like `allowPrivateSubnet` to override this check. This will force our users to reckon with canary-specific vpc considerations at synth time.

Just spitballing here, I'm not very familiar with vpcs and their use cases so I'm happy to have some dialogue about this.

Reply:
Yah, there's not a good way for us to check if the VPCe are available, so this might be a nice synth time validation. This was largely inspired by Lambda (since Synth's VPC is all just Lambda VPC), but I think this is a Synth specific caveat so worth checking.

Reply:
It's a bit roundabout, we now check if all the "selected" subnets are "isolated" which is VPC verbage for no internet access (private w/ no IGW or NAT) and say how to fix it.